### PR TITLE
CDAP-8850 Handle NoSuchMethodException

### PR DESCRIPTION
**Pull request by deepakwadhwani**
- Handle NoSuchMethodException when getting the main method. 
- call invoke independently so exceptions [like missing dependency] are clearly thrown.

### CODE REVIEW

File: cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala [hunk 1]
```diff
@@ -135,12 +128,16 @@ object SparkMainWrapper {
     }
   }
 
-  private def getMainMethod(obj: Class[_]): Option[Method] = {
+  private def getMainMethod(obj: Class[_]): Method = {
     try {
       val mainMethod = obj.getDeclaredMethod("main", classOf[Array[String]])
-      if (Modifier.isStatic(mainMethod.getModifiers)) Some(mainMethod) else None
+      if (Modifier.isStatic(mainMethod.getModifiers)) mainMethod else throw new IllegalArgumentException("Static "
```
Comment (chtyim) [inline cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala]: It'll be cleaner if you do

```
if (!Modifier.isStatic(...)) {
  throw new IllegalArgumentException(....);
}
mainMethod
```

Comment (deepakwadhwani) [reply cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala]: @chtyim can you please check again.

File: cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala [hunk 2]
```diff
@@ -135,12 +128,19 @@ object SparkMainWrapper {
...
   }
 
-  private def getMainMethod(obj: Class[_]): Option[Method] = {
+  private def getMainMethod(obj: Class[_]): Method = {
     try {
       val mainMethod = obj.getDeclaredMethod("main", classOf[Array[String]])
-      if (Modifier.isStatic(mainMethod.getModifiers)) Some(mainMethod) else None
+      if (!Modifier.isStatic(mainMethod.getModifiers)) {
+        throw new IllegalArgumentException("Static "
+          + "modifiers not used for main method of " + obj.getName)
```
Comment (chtyim) [inline cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala]: Can be on one line, right?

Comment (deepakwadhwani) [reply cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SparkMainWrapper.scala]: oops, yes. updated again.

**Comment by chtyim** [comment]
LGTM. Please squash before merging.

**Comment by deepakwadhwani** [comment]
Build: no new failures: https://builds.cask.co/browse/CDAP-RUT1067-1

**Pull request merged**